pip/commands/install.py

```diff
@@ -159,6 +159,14 @@ def __init__(self, *args, **kw):
 
         cmd_opts.add_option(cmdoptions.no_clean())
 
+        cmd_opts.add_option(
+            '--require-hashes',
```

Comment:
Seems reasonable and not too hard, looking at the code. I did all the operative stuff in RequirementSet, after all. But perhaps it could be a separate PR, later, since I'm about out of time to work on this. Also, nothing supports --hash options on the commandline yet, which is a sad hole in the story. That will need either some serious monkeying around with optparse to support interspersed args and options or else your aforementioned switch to `click`.